tests/Unit/Domain/DomainCreators/Test_Shell.cpp

```diff
@@ -417,3 +429,111 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Shell.Factory.ShellWedges",
+      {{-x_interior, zero, zero}}};
+  const DomainCreators::Shell<Frame::Inertial> shell{
+      inner_radius - 0.1,    outer_radius + 0.1,  refinement_level,
+      grid_points_r_angular, use_equiangular_map, aspect_ratio,
+      use_logarithmic_map,   which_wedges,        radial_block_layers};
+  auto domain = shell.create_domain();
+  const auto blogical_coords =
+      block_logical_coordinates(domain, interior_inertial_coords);
+  for (size_t s = 0; s < expected_block_ids.size(); ++s) {
+    CHECK(blogical_coords[s].id.get_index() == expected_block_ids[s]);
+  }
+  size_t element_count = 0;
+  for (const auto& block : domain.blocks()) {
+    const auto initial_ref_levs = shell.initial_refinement_levels()[block.id()];
+    const std::vector<ElementId<3>> element_ids =
+        initial_element_ids(block.id(), initial_ref_levs);
+    for (const auto& element_id : element_ids) {
+      const auto element = create_initial_element(element_id, block);
+      //Creating elements through InitialRefinement creates Elements in all
+      //three logical dimensions. It is sufficient to only check the elements
+      //in the radial direction lying along a ray at a fixed angle.
+      //This is done by getting the desired Elements through their SegmentIds.
+      if (element_id.segment_ids()[0] == SegmentId{refinement_level, 0} and
+          element_id.segment_ids()[1] == SegmentId{refinement_level, 0}) {
+        element_count++;
+        const auto map = ElementMap<3, Frame::Inertial>{
+            element_id, block.coordinate_map().get_clone()};
+        const tnsr::I<double, 3, Frame::Logical> logical_point(
+            std::array<double, 3>{{0.0, 0.0, 1.0}});
+        CHECK(magnitude(map(logical_point)).get() ==
+              approx(x_boundary[element_count]));
+      }
+    }
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Shell.RadialBlockLayers",
+                  "[Domain][Unit]") {
+  const auto log_shell =
+      test_factory_creation<DomainCreator<3, Frame::Inertial>>(
+          "  Shell:\n"
+          "    InnerRadius: 1\n"
+          "    OuterRadius: 3\n"
+          "    InitialRefinement: 2\n"
+          "    InitialGridPoints: [2,3]\n"
+          "    UseEquiangularMap: false\n"
+          "    AspectRatio: 2.0        \n"
+          "    UseLogarithmicMap: true\n"
+          "    RadialBlockLayers: 6\n");
+   const auto shell = test_factory_creation<DomainCreator<3, Frame::Inertial>>(
+      "  Shell:\n"
+      "    InnerRadius: 1\n"
+      "    OuterRadius: 3\n"
+      "    InitialRefinement: 2\n"
+      "    InitialGridPoints: [2,3]\n"
+      "    UseEquiangularMap: false\n"
+      "    AspectRatio: 2.0        \n"
+      "    UseLogarithmicMap: false\n"
+      "    RadialBlockLayers: 4\n");
+
+  test_radial_block_layers(1.0, 10.0, 0, true, 8, {0, 1, 2, 3, 4, 5, 6, 7, 7});
+  test_radial_block_layers(2.0, 9.0, 0, false, 8, {0, 1, 2, 3, 3, 4, 5, 6, 7});
```

Comment:
Where did these numbers (the lists at the end) come from?

Reply:
ping